src/gallia/dumpcap.py

```diff
@@ -144,11 +144,15 @@ def _can_cmd(
         return args
 
     @staticmethod
-    async def _eth_cmd(target_ip: str) -> list[str]:
+    async def _eth_cmd(target_ip: str) -> list[str] | None:
+        if len(target_ip) == 0:
+            return None
```

Comment:
I personally still consider a *failing and enabled* dumpcap a *critical error*, since expected logs will be missing in this case. So, the commit a40927de437a488e65105f477c03fc17070d3734 needs to be discussed: @peckto @fkglr. I personally would keep the current state as is and fail hard in this case. If you do not need dumpcap, it can be disabled (also via `gallia.toml`).

If there are valid reasons why `len(target_ip)` could be `0`, those should be handled by the caller, I guess.

Reply:
I think you are misunderstanding.

The behavior you are describing is already on master: https://github.com/Fraunhofer-AISEC/gallia/blob/master/src/gallia/command/base.py#L552

This commit just handles incorrect dumpcap invocations, such as when a target like `unix-lines///tmp.sock` is given...
The main problem here is that the `else` branch is too broad: https://github.com/Fraunhofer-AISEC/gallia/blob/master/src/gallia/dumpcap.py#L64

Reply:
> The behavior you are describing is already on master: https://github.com/Fraunhofer-AISEC/gallia/blob/master/src/gallia/command/base.py#L552

You are right, I missed that and it is offtopic for this review here. However, I think this topic is still relevant and I will create an issue for this to discuss this further.

> This commit just handles incorrect dumpcap invocations, such as when a target like unix-lines///tmp.sock is given...

Yes, this is a special case in the function that handles arguments for ethernet setups. In the described case the `_eth_cmd` is called with wrong arguments. As mentioned, the main problem should be handled by the caller (i.e. the too broad else branch).

Reply:
>As mentioned, the main problem should be handled by the caller (i.e. the too broad else branch).

Both, no? So the else branch shouldn't be too broad (feel free to create MR), but the `_eth_cmd` can and should also do some basic sanity checking. The current approach with returning None matches e.g. [the behavior of `_can_cmd`](https://github.com/Fraunhofer-AISEC/gallia/blob/master/src/gallia/dumpcap.py#L135-L137).

Reply:
And to state the (not so?) obvious: an empty string `target_ip` does not raise an error here: https://github.com/Fraunhofer-AISEC/gallia/blob/tp-shenanigangs/src/gallia/dumpcap.py#L151-L155

Reply:
> Both, no? So the else branch shouldn't be too broad (feel free to create MR), but the _eth_cmd can and should also do some basic sanity checking. 

A sanity check of the arguments makes sence. However, the URI scheme is already checked in the caller and as mentioned this check is not complete. Let's fix this there.

> an empty string target_ip does not raise an error here

Yes, and I still do not see a valid reason for this. If there is no `target_ip` set, it makes no sense to call `_eth_cmd()`.

Reply:
will adapt once you merged #489

Reply:
Adapted the behavior of `_eth_cmd` to match `_can_cmd` and make it return `None` on error.

Considering this still is the "default" due to `case _:` I think it remains a reasonable change.

Reply:
Tested Branch. asyncio and teardown errors have been fixed. 
Only one error remains with python3.11:
`Exception ignored in: <function StreamWriter.__del__ at 0x7ff712588c20>
Traceback (most recent call last):
  File "/usr/lib/python3.11/asyncio/streams.py", line 395, in __del__
  File "/usr/lib/python3.11/asyncio/streams.py", line 343, in close
  File "/usr/lib/python3.11/asyncio/selector_events.py", line 860, in close
  File "/usr/lib/python3.11/asyncio/base_events.py", line 761, in call_soon
  File "/usr/lib/python3.11/asyncio/base_events.py", line 519, in _check_closed
RuntimeError: Event loop is closed`

Reply:
I understand you are OK with the changes in 37a368fbe883ccfa99e7c9ab49875aa56dba9eac ?

Care to resolve this conversation?